File: back/DevC/src/main/java/com/back/devc/global/initData/CategoryInitData.java
package com.back.devc.global.initData;

// 테스트를 수행하기 위해서 일단 카테고리 임시 추가
import com.back.devc.domain.post.category.entity.Category;
import com.back.devc.domain.post.category.repository.CategoryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.ApplicationRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class CategoryInitData {

    private final CategoryRepository categoryRepository;

    @Bean
    public ApplicationRunner init() {
        return args -> {

            if (categoryRepository.count() > 0) return;

            List<String> names = List.of("tech", "job-market", "trend","free");

            for (String name : names) {
                Category c = new Category(name);
                categoryRepository.save(c);
            }
        };
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/global/initData/CategoryInitData.kt
package com.back.devc.global.initData

import com.back.devc.domain.post.category.entity.Category
import com.back.devc.domain.post.category.repository.CategoryRepository
import org.springframework.boot.ApplicationRunner
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component

@Component
class CategoryInitData(
    private val categoryRepository: CategoryRepository
) {

    @Bean
    fun init(): ApplicationRunner {
        return ApplicationRunner {
            val names = listOf("tech", "job-market", "trend", "free", "discussion")

            for (name in names) {
                if (categoryRepository.existsByName(name)) {
                    continue
                }

                val category = Category(name)
                categoryRepository.save(category)
            }
        }
    }
}